### ISSUE TITLE
Use third party CLI tool for various jOOQ CLIs

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#9625]
In the near future, we will be offering more CLI to jOOQ users. These include:

- #7269 `ParserCLI` (shipped in 3.11)
- #7621 A SQL CLI
- #9624 A diff CLI

Other possible CLIs with no issue number yet:

- A code generation CLI
- A migration CLI

With these many CLIs it may make sense to think about:

- Creating a new `jOOQ-cli` module. Existing CLIs like the `ParserCLI` could be moved here, incompatibly.
- Optionally offering a new `assembly:single` build for the CLI module for users who don't care about the jOOQ Java API but just want to use the CLI. This could be done with some shell scripts as well, which manage class paths for users.
- Using a third party library like https://github.com/remkop/picocli (to be evaluated, other options are possible)

**Comment by knutwannheden** [comment]
Yes, this would be nice! Picocli also offers [integration with JLine](https://github.com/remkop/picocli/tree/master/picocli-shell-jline3), which would be of particular interest to us, as JLine provides the features for interactive shell applications (like for example command history).

**Comment by lukaseder** [comment]
> like for example command history

I've thought about this before. `ParserCLI` should have a command history, but we shouldn't spend a ton of time implementing one.
